### ISSUE TITLE
Regression in SQLite code generation for unnamed composite pkeys on JDK 9+ with the jOOQ Open Source Edition

### ISSUE THREAD

**Issue by Ubehebe** [jOOQ/jOOQ#12118]
Minimal repro:

```sql
CREATE TABLE aliases (
  id text,
  alias text,
  PRIMARY KEY (id, alias)
);
```

Put that into a SQLite database (I'm using version 3.36).

### Expected behavior
jOOQ v3.14.11 generates the following code for the pkey:

```java
/**
 * A class modelling foreign key relationships and constraints of tables in 
 * the default schema.
 */
@SuppressWarnings({ "all", "unchecked", "rawtypes" })
public class Keys {

    // -------------------------------------------------------------------------
    // UNIQUE and PRIMARY KEY definitions
    // -------------------------------------------------------------------------

    public static final UniqueKey<AliasesRecord> PK_ALIASES = Internal.createUniqueKey(Aliases.ALIASES, DSL.name("pk_aliases"), new TableField[] { Aliases.ALIASES.ID, Aliases.ALIASES.ALIAS }, true);
}
```

### Actual behavior
When I upgrade to jOOQ v3.15.0, the same SQLite db generates the following code for the pkey:

```java
/**
 * A class modelling foreign key relationships and constraints of tables in the
 * default schema.
 */
@SuppressWarnings({ "all", "unchecked", "rawtypes" })
public class Keys {

    // -------------------------------------------------------------------------
    // UNIQUE and PRIMARY KEY definitions
    // -------------------------------------------------------------------------

    public static final UniqueKey<AliasesRecord> _ = Internal.createUniqueKey(Aliases.ALIASES, DSL.name(""), new TableField[] { Aliases.ALIASES.ID, Aliases.ALIASES.ALIAS }, true);
}
```

This does not compile with javac 9+ because `_` is no longer a valid identifier.

### Versions

- jOOQ: v3.14.11 (good) -> v3.15.0 (bad)
- Java: 14
- Database (include vendor): SQLite 3.36
- OS: MacOS/Linux

**Comment by lukaseder** [comment]
Thanks a lot for your report. Will look into this right away.

**Comment by lukaseder** [comment]
This generates `ALIASES__` for me. Can you check if you're really using the correct versions of the jOOQ runtime library *and* the code generator? Otherwise, would you mind creating an MCVE to help reproduce this? Template here: https://github.com/jOOQ/jOOQ-mcve

**Comment by Ubehebe** [comment]
Reproed on both Linux and MacOS. The linked MCVE generates `public static final UniqueKey<AliasesRecord> _` with jOOQ v3.15.0 and generates `public static final UniqueKey<AliasesRecord> PK_ALIASES` with jOOQ v3.14.11.

(By the way, thank you very much for all your work on jOOQ.)

**Comment by lukaseder** [comment]
I'm assuming you mean this MCVE: https://github.com/Ubehebe/jOOQ-mcve. Thanks a lot for providing that. Will look into it right away.

**Comment by lukaseder** [comment]
I can reproduce the problem with the MCVE, thanks again.

**Comment by lukaseder** [comment]
Oh, I see. This is because of this change here: https://github.com/jOOQ/jOOQ/issues/11286. The change produced a regression in the jOOQ Open Source Edition. It works fine in the jOOQ Professional Edition. The change was using previously commercial-only code from https://github.com/jOOQ/jOOQ/issues/5538.

It suffices to make that code available to all editions.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.16.0 and 3.15.1 (https://github.com/jOOQ/jOOQ/issues/12147)

**Comment by lukaseder** [comment]
The fix still leaves a problem in the `ScalaGenerator`, which can't handle an identifier ending on an underscore followed by a `:`. We need to insert the usual whitespace.
